### ISSUE TITLE
Error installing

### ISSUE THREAD

**Issue by han-solo318** [MiragianCycle/OVIWrite#12]
When opening up neovim after attempting to install, I ran into the following error:
```
Erreur détectée en traitant /home/hansolo318/.config/nvim/init.lua :                                                                     
Failed to load `plugins.catppuccin`                                                                                                      
                                                                                                                                         
...local/share/nvim/lazy/lazy.nvim/lua/lazy/core/plugin.lua:184: attempt to concatenate local 'err' (a table value)                      
                                                                                                                                         
# stacktrace:                                                                                                                            
  - .config/nvim/lua/config/init.lua:50                                                                                                  
  - .config/nvim/init.lua:1                                                                                                              
Invalid plugin spec {                                                                                                                    
  select_template = <function 1>                                                                                                         
}                                                                                                                                        
Appuyez sur ENTRÉE ou tapez une commande pour continuer
```

When I opened up `catppuccin.lua` in my plugins folder, I saw the following:

```lua
return 
{ "catppuccin/nvim"},
{
name = "catppuccin", priority = 1000 
}
```

I tried changing it to this, and reran nvim.

```lua
return 
{ "catppuccin/nvim",
  name = "catppuccin", 
  priority = 1000 
}
```

However, I then ran into the following error:

```
Erreur détectée en traitant /home/hansolo318/.config/nvim/init.lua :                                                                     
Invalid plugin spec {                                                                                                                    
  select_template = <function 1>                                                                                                         
}                                                                                                                                        
Appuyez sur ENTRÉE ou tapez une commande pour continuer 
```

I couldn't figure out how to solve this one. Is there any way that you could help? Let me know if there's any relevant info I'm missing.

I'm running Linux Mint 21.3 XFCE, with nvim v0.10.0, npm v10.8.2, git v2.34.1, and I followed instructions from the installation.md file using my package manager.  I installed nvim using appimage, and I'm not sure if that could be the cause of my issues.

Please let me know if you need more information, or if I should go somewhere else for this. Sorry if this is the wrong channel for submitting requests like this.

I'd also like to add my checkhealth report:

<details>
<summary>Checkhealth report</summary>

```

==============================================================================
lazy: require("lazy.health").check()

lazy.nvim ~
- {lazy.nvim} version `11.14.1`
- OK {git} `version 2.34.1`
- OK no existing packages found by other package managers
- OK packer_compiled.lua not found
- ERROR Issues were reported when loading your specs:
- WARNING {nvim-tree.lua}: setting a table to `Plugin.config` is deprecated. Please use `Plugin.opts` instead
- ERROR Invalid plugin spec {
- ERROR   select_template = <function 1>
- ERROR }

luarocks ~
- checking `luarocks` installation
- OK no plugins require `luarocks`, so you can ignore any warnings below
- OK {luarocks} `/usr/local/bin/luarocks 3.11.1`
- WARNING `lua` version `5.1` needed, but found `Lua 5.4.7  Copyright (C) 1994-2024 Lua.org, PUC-Rio`
- WARNING {lua5.1} or {lua} or {lua-5.1} version `5.1` not installed

==============================================================================
noice: require("noice.health").check()

noice.nvim ~
- OK **Neovim** >= 0.9.0
- OK You're using a GUI that should work ok
- OK **vim.go.lazyredraw** is not enabled
- OK **nvim-notify** is installed
- OK **TreeSitter vim** parser is installed
- WARNING **TreeSitter regex** parser is not installed. Highlighting of the cmdline for regex might be broken
- OK **TreeSitter lua** parser is installed
- OK **TreeSitter bash** parser is installed
- OK **TreeSitter markdown** parser is installed
- OK **TreeSitter markdown_inline** parser is installed
- OK `vim.notify` is set to **Noice**
- OK `vim.lsp.handlers["textDocument/hover"]` is set to **Noice**
- OK `vim.lsp.handlers["textDocument/signatureHelp"]` is set to **Noice**
- OK `vim.lsp.handlers["window/showMessage"]` is set to **Noice**
- OK `vim.lsp.util.convert_input_to_markdown_lines` is set to **Noice**
- OK `vim.lsp.util.stylize_markdown` is set to **Noice**

==============================================================================
nvim: require("nvim.health").check()

Configuration ~
- OK no issues found

Runtime ~
- OK $VIMRUNTIME: /tmp/.mount_nvimdbu6Dc/usr/share/nvim/runtime

Performance ~
- OK Build type: Release

Remote Plugins ~
- OK Up to date

terminal ~
- key_backspace (kbs) terminfo entry: `key_backspace=\177`
- key_dc (kdch1) terminfo entry: `key_dc=\E[3~`
- $COLORTERM="truecolor"

External Tools ~
- OK ripgrep 13.0.0 (/usr/bin/rg)

==============================================================================
nvim-treesitter: require("nvim-treesitter.health").check()

Installation ~
- WARNING `tree-sitter` executable not found (parser generator, only needed for :TSInstallFromGrammar, not required for :TSInstall)
- OK `node` found v22.5.1 (only needed for :TSInstallFromGrammar)
- OK `git` executable found.
- OK `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
  Version: cc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
- OK Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

OS Info:
{
  machine = "x86_64",
  release = "5.15.0-122-generic",
  sysname = "Linux",
  version = "#132-Ubuntu SMP Thu Aug 29 13:45:52 UTC 2024"
} ~

Parser/Features         H L F I J
  - bash                ✓ ✓ ✓ . ✓
  - c                   ✓ ✓ ✓ ✓ ✓
  - lua                 ✓ ✓ ✓ ✓ ✓
  - markdown            ✓ . ✓ ✓ ✓
  - markdown_inline     ✓ . . . ✓
  - org                 ✓ . ✓ . ✓
  - python              ✓ ✓ ✓ ✓ ✓
  - query               ✓ ✓ ✓ ✓ ✓
  - vim                 ✓ ✓ ✓ . ✓
  - vimdoc              ✓ . . . ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang} ~

==============================================================================
provider.clipboard: require("provider.clipboard.health").check()

Clipboard (optional) ~
- OK Clipboard tool found: xclip

==============================================================================
provider.node: require("provider.node.health").check()

Node.js provider (optional) ~
- Node.js: v22.5.1
  
- Nvim node.js host: /home/hansolo318/.nvm/versions/node/v22.5.1/lib/node_modules/neovim/bin/cli.js
- OK Latest "neovim" npm/yarn/pnpm package is installed: 5.2.1
  

==============================================================================
provider.perl: require("provider.perl.health").check()

Perl provider (optional) ~
- WARNING "Neovim::Ext" cpan module is not installed
  - ADVICE:
    - See :help |provider-perl| for more information.
    - You may disable this provider (and warning) by adding `let g:loaded_perl_provider = 0` to your init.vim
- WARNING No usable perl executable found

==============================================================================
provider.python: require("provider.python.health").check()

Python 3 provider (optional) ~
- `g:python3_host_prog` is not set. Searching for python3 in the environment.
- Executable: /usr/bin/python3
- Python version: 3.10.12
- pynvim version: 0.4.2
- OK Latest pynvim is installed.

Python virtualenv ~
- OK no $VIRTUAL_ENV

==============================================================================
provider.ruby: require("provider.ruby.health").check()

Ruby provider (optional) ~
- Ruby: ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x86_64-linux-gnu]
- WARNING `neovim-ruby-host` not found.
  - ADVICE:
    - Run `gem install neovim` to ensure the neovim RubyGem is installed.
    - Run `gem environment` to ensure the gem bin directory is in $PATH.
    - If you are using rvm/rbenv/chruby, try "rehashing".
    - See :help |g:ruby_host_prog| for non-standard gem installations.
    - You may disable this provider (and warning) by adding `let g:loaded_ruby_provider = 0` to your init.vim

==============================================================================
telescope: require("telescope.health").check()

Checking for required plugins ~
- OK plenary installed.
- OK nvim-treesitter installed.

Checking external dependencies ~
- OK rg: found ripgrep 13.0.0
- OK fd: found fd 8.3.1

===== Installed extensions ===== ~

Telescope Extension: `notify` ~
- No healthcheck provided

==============================================================================
vim.lsp: require("vim.lsp.health").check()

- LSP log level : WARN
- Log path: /home/hansolo318/.local/state/nvim/lsp.log
- Log size: 0 KB

vim.lsp: Active Clients ~
- No active clients

vim.lsp: File watcher ~
- file watching "(workspace/didChangeWatchedFiles)" disabled on all clients

==============================================================================
vim.treesitter: require("vim.treesitter.health").check()

- Nvim runtime ABI version: 14
- OK Parser: org                  ABI: 14, path: /home/hansolo318/.local/share/nvim/lazy/nvim-treesitter/parser/org.so
- OK Parser: query                ABI: 14, path: /home/hansolo318/.local/share/nvim/lazy/nvim-treesitter/parser/query.so
- OK Parser: vimdoc               ABI: 14, path: /home/hansolo318/.local/share/nvim/lazy/nvim-treesitter/parser/vimdoc.so
- OK Parser: org                  ABI: 14, path: /home/hansolo318/.local/share/nvim/lazy/orgmode/parser/org.so
- OK Parser: bash                 ABI: 14, path: /tmp/.mount_nvimdbu6Dc/usr/lib/nvim/parser/bash.so
- OK Parser: c                    ABI: 14, path: /tmp/.mount_nvimdbu6Dc/usr/lib/nvim/parser/c.so
- OK Parser: lua                  ABI: 14, path: /tmp/.mount_nvimdbu6Dc/usr/lib/nvim/parser/lua.so
- OK Parser: markdown             ABI: 14, path: /tmp/.mount_nvimdbu6Dc/usr/lib/nvim/parser/markdown.so
- OK Parser: markdown_inline      ABI: 14, path: /tmp/.mount_nvimdbu6Dc/usr/lib/nvim/parser/markdown_inline.so
- OK Parser: python               ABI: 14, path: /tmp/.mount_nvimdbu6Dc/usr/lib/nvim/parser/python.so
- OK Parser: query                ABI: 14, path: /tmp/.mount_nvimdbu6Dc/usr/lib/nvim/parser/query.so
- OK Parser: vim                  ABI: 14, path: /tmp/.mount_nvimdbu6Dc/usr/lib/nvim/parser/vim.so
- OK Parser: vimdoc               ABI: 14, path: /tmp/.mount_nvimdbu6Dc/usr/lib/nvim/parser/vimdoc.so
```

</details>

**Comment by sbishop333** [comment]
Hi,

It looks like you have already solved the first error. Your description
is essentially how I did it.

The second error indicates a problem with a function that loads a menu
to select a template for Latex, Markdown, or Orgmode documents. I have
not had time to find the problem, but you can stop the error on startup
by renaming ~/.config/nvim/lua/writer_templates/init.lua to something
like init.lua.txt. Lazy then ignores the file during loading and the
error message goes away. 

As for your installation being in appimage format, I'm not sure if
that's a problem since you are able to start OVIWrite and get the error
messages below. I use a standard installation based on code taken from
the neovim.io website. The nvim executable installs in /opt/nvim-
linux64/bin directory. Add that to your $PATH variable and you should
be able to start Neovim from anywhere using the terminal.

Good luck!

Steve B.

**Comment by han-solo318** [comment]
Thanks for the reply. Your solution did work, but I'm still not able to get it up and running properly. Now, I'm having an issue where only a few of the plugins will load.
![Capture d’écran_2024-09-28_19-49-37](https://github.com/user-attachments/assets/bc0373f8-b077-44bf-b585-d2292d4fe599)

I think the culprit is `markdown-preview.nvim` based on what happens when I try to sync:
![Capture d’écran_2024-09-28_19-52-21](https://github.com/user-attachments/assets/cd6d4fe0-3a04-4943-855e-d13bb1a7adbc)

I've spent several hours trying to fix it. First, I tried switching the `build` command to be with `yarn` instead of `npm`, but it didn't seem to help. I've tried deleting the `.local/share/nvim` directory and letting it rebuild, but to no avail. All the packages are installed, but only a few of them will load. I updated neovim, made sure yarn and npm are up to date, and looked around everywhere for a solution.

I saw some people on this thread https://github.com/iamcco/markdown-preview.nvim/issues/616 were having a similar issue, but none of the fixes worked for me. Not sure if I should be reporting this upstream or if the issue is with my configuration.

**Comment by sbishop333** [comment]
It appears that many of the plugins are not configured to load on
startup; others are configured to load when certain triggers are fired.
Markdown-preview.nvim loads when a file of filetype "markdown" is
opened--at least it does on my system. Whether or not it actually works
as expected is another issue! The plugin loads properly when I open a
Markdown file, but when I key in the MarkdownPreview command, the
node.js version flashes in the message box but nothing else happens.
Perhaps markdown-previw.nvim requires an older version of node.js? 

Fortunately for me, I write everything in LaTeX, and that works just
fine for me, including previewing the output in my pdf viewer. However,
the Vimtex plugin was not configured to load automatically when a tex
file is opened! I had to add "lazy = false" to the vimtex.lua file
under plugins directory and I had to do the same for goyo.vim. The most
recent version of OVIWrite, cloned per the web site instructions from
the Github repository, has a new autocommand that opens the pdf of a
latex file when the the file is loaded. Very convenient!

Hope there is something useful here!

Good luck!

SB
